src/main/java/org/opensearch/knn/indices/ModelMetadata.java

```diff
@@ -321,11 +379,16 @@ public static ModelMetadata getMetadataFromSourceMap(final Map<String, Object> m
         Object description = modelSourceMap.get(KNNConstants.MODEL_DESCRIPTION);
         Object error = modelSourceMap.get(KNNConstants.MODEL_ERROR);
         Object trainingNodeAssignment = modelSourceMap.get(KNNConstants.MODEL_NODE_ASSIGNMENT);
+        Object methodComponentContext = modelSourceMap.get(KNNConstants.MODEL_METHOD_COMPONENT_CONTEXT);
```

Comment:
I dont think this will work. Because we are indexing the methodComponentContext as a string, we are going it will be represented as a string in the source map. We should call from string on this to get the correct value.

Add a test for this as well.